crates/blockchain/mempool.rs

```diff
@@ -206,6 +206,34 @@ fn validate_transaction(tx: &Transaction, store: Store) -> Result<(), MempoolErr
         }
     }
 
+    let maybe_sender_acc_info = store.get_account_info(header_no, tx.sender())?;
+
+    if let Some(sender_acc_info) = maybe_sender_acc_info {
+        if tx.nonce() < sender_acc_info.nonce {
+            return Err(MempoolError::InvalidNonce);
+        }
+
+        let effective_gas_price: U256 = tx
+            .effective_gas_price(header.base_fee_per_gas)
+            .ok_or(MempoolError::InvalidTxGasvalues)?
+            .into();
+        let gas_limit = U256::from(tx.gas_limit());
+        let tx_cost: U256 = effective_gas_price * gas_limit + tx.value();
```

Comment:
Let's use saturating add, since this can overflow and tx.value() is passed by the user.

Reply:
Switched to saturating add and mul here [b988558](https://github.com/lambdaclass/lambda_ethereum_rust/pull/1125/commits/b988558443dd54585642e3b720454c2422f56292)